tests/RobotFramework/tests/tedge/tedge_http.robot

```diff
@@ -0,0 +1,93 @@
+*** Settings ***
+Resource            ../resources/common.resource
+Library             Cumulocity
+Library             ThinEdgeIO
+
+Suite Setup         Custom Setup
+Test Teardown       Test Teardown
+
+Test Tags           theme:cli    theme:http    theme:childdevices
+
+
+*** Test Cases ***
+Sanity check: No HTTP service on a child device
+    Execute Command    curl http://localhost:8000/tedge/entity-store/v1/entities    exp_exit_code=7
+
+Listing entities from a child device
+    ${entities}=    Execute Command    tedge http get /tedge/entity-store/v1/entities
+    Should Contain    ${entities}    device/main//
+    Should Contain    ${entities}    device/${CHILD_SN}//
+
+Updating entities from a child device
+    Execute Command
+    ...    tedge http post /tedge/entity-store/v1/entities '{"@topic-id": "device/${CHILD_SN}/service/watchdog", "@type": "service", "@parent": "device/${CHILD_SN}//"}'
+    ${entity}=    Execute Command    tedge http get /tedge/entity-store/v1/entities/device/${CHILD_SN}/service/watchdog
+    Should Contain    ${entity}    "@topic-id":"device/${CHILD_SN}/service/watchdog"
+    Should Contain    ${entity}    "@parent":"device/${CHILD_SN}//"
+    Should Contain    ${entity}    "@type":"service"
+    Execute Command    tedge http delete /tedge/entity-store/v1/entities/device/${CHILD_SN}/service/watchdog
+    Execute Command
+    ...    tedge http get /tedge/entity-store/v1/entities/device/${CHILD_SN}/service/watchdog
+    ...    exp_exit_code=1
+
+Accessing c8y from a child device
+    ${external_id}=    Execute Command
+    ...    tedge http get /c8y/identity/externalIds/c8y_Serial/${PARENT_SN} | jq .externalId
+    Should Be Equal    ${external_id}    "${PARENT_SN}"\n
+
+Accessing file-transfer from a child device
+    Execute Command    echo "source file content" >/tmp/source-file.txt
+    Execute Command    tedge http put /tedge/file-transfer/target --file /tmp/source-file.txt --content-type text/plain
+    ${content}=    Execute Command    tedge http get /tedge/file-transfer/target
+    Should Be Equal    ${content}    source file content\n\n
```

Comment:
Once the extra `\n` in the output from `tedge http get` is addressed then it might be worthwhile refactoring the test to be clear on the exact format expected (as the tester might not realize that `echo` adds a newline char by default).

So using `printf` is more explicit compared to `echo`, and it won't add a newline char to the file.

```suggestion
    Execute Command    printf "source file content" >/tmp/source-file.txt
    Execute Command    tedge http put /tedge/file-transfer/target --file /tmp/source-file.txt --content-type text/plain
    ${content}=    Execute Command    tedge http get /tedge/file-transfer/target
    Should Be Equal    ${content}    source file content
```

Reply:
Fixed: https://github.com/thin-edge/thin-edge.io/pull/3357/commits/8b4dda4d260498352f1c4057c65a02777cb0fc3b